### PR TITLE
Add the possibity to include metadata in the logs

### PR DESCRIPTION
**Pull request by jcague**
With this change both publish() and subscribe() methods in clients will accept a new field in options called metadata. 

Metadata is an Object with keys and values, both should be strings. Below you can find an example:
```
{metadata: {type: 'publisher', external_id: 'id1'}}
```

All fields inside metadata will be printed in logs in Erizo in this PR. And we will add them to ErizoController logs too.

### CODE REVIEW

File: erizo_controller/erizoJS/erizoJSController.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ exports.ErizoJSController = function () {
     /*
      * Enables/Disables slideshow mode for a subscriber
      */
-    setSlideShow = function (slideShowMode, from, to){
+    setSlideShow = function (slideShowMode, from, to) {
```
Comment (lodoyun) [inline erizo_controller/erizoJS/erizoJSController.js]: I think minor cosmetic problems like this in the future should be grouped in another PR, they add too much noise and create  merge conflicts when there shouldn't be.

File: erizo_controller/erizoClient/src/Room.js [hunk 2]
```diff
@@ -720,7 +724,8 @@ Erizo.Room = function (spec) {
                 // 1- Subscribe to Stream
 
                 if (that.p2p) {
-                    sendSDPSocket('subscribe', {streamId: stream.getID()});
+                    sendSDPSocket('subscribe', {streamId: stream.getID(),
```
Comment (lodoyun) [inline erizo_controller/erizoClient/src/Room.js]: Does this have any effect on p2p streams as it is now?

File: erizo/src/erizo/logger.h [hunk 3]
```diff
@@ -25,6 +25,38 @@
 #include <log4cxx/logger.h>
 #include <log4cxx/helpers/exception.h>
 
+#include <map>
+#include <string>
+#include <utility>
+
+class LogContext {
```
Comment (lodoyun) [inline erizo/src/erizo/logger.h]: :+1:

File: erizo_controller/erizoClient/src/Room.js (hunk 2 above)
Comment (jcague) [inline erizo_controller/erizoClient/src/Room.js]: nope, but it will have if we add logs to ErizoController in the short future (about an hour 😄 )

Comment (lodoyun) [inline erizo_controller/erizoClient/src/Room.js]: 👏

File: extras/basic_example/public/script.js [hunk 4]
```diff
@@ -93,15 +93,15 @@ window.onload = function () {
...
+            room.subscribe(stream, {slideShowMode: slideShowMode, metadata: {type: 'subscriber'}});
             stream.addEventListener('bandwidth-alert', cb);
           }
         }
       };
 
       room.addEventListener('room-connected', function (roomEvent) {
 
-        room.publish(localStream, {maxVideoBW: 300});
+        room.publish(localStream, {maxVideoBW: 300, metadata: {type: 'publisher'}});
```
Comment (aalonsog) [inline extras/basic_example/public/script.js]: Ok, with this I understand why metadata is desired :)

**Comment by jcague** [comment]
@lodoyun completely agree. I updated the description with the changes in the public API.

**Pull request merged**